Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -820,6 +861,79 @@ public Task Sort( DataGridColumn<TItem> column, SortDirection? sortDirection = n
         return Task.CompletedTask;
     }
 
+
+    /// <summary>
+    /// Replaces the current sorting with the given sort order
+    /// </summary>
+    /// <param name="columns">Columns used for sorting</param>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    /// <remarks>
+    /// Note that <see cref="DataGridColumn{TItem}.Sortable"/> and <see cref="Sortable"/> must be enabled to be able to sort!
+    /// If more than one column is specified, <see cref="SortMode"/> must be <see cref="DataGridSortMode.Multiple"/>
+    /// </remarks>
+    public async Task ApplySorting(params DataGridSortColumn[] columns ) {
+        if ( columns.Length > 1 && SortMode == DataGridSortMode.Single )
+        {
+            throw new ArgumentException(
+                $"Cannot sort on multiple columns when {nameof(SortMode)} is {nameof(DataGridSortMode.Single)}." );
+        }
+
+        if ( SortMode == DataGridSortMode.Single )
+        {
+            if ( columns.Length == 1 )
+            {
+                await Sort( columns[0].Field, columns[0].SortDirection );
+            }
+            else if ( SortByColumns.Count == 1 )

```

Comment:
Can you explain?
You've done a sanity check at the top for columns > 1 + Single. 
How would the **else if** trigger here and why? If it was empty? Does that make sense?

Reply:
I added code comments to clarify the if/else statement. Basically, the ELSE branch checks whether the DataGrid is already sorted by a specific column. If this is the case, the sorting must be **reset** here (the user has not passed any sorting columns).

Is this okay?